tests/avifmetadatatest.c

```diff
@@ -0,0 +1,221 @@
+// Copyright 2022 Google LLC. All rights reserved.
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <assert.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "avif/avif.h"
+
+//------------------------------------------------------------------------------
+
+// ICC color profiles are not checked by libavif so the content does not matter.
+// This is a truncated widespread ICC color profile.
+static const uint8_t sampleICC[] = { 0x00, 0x00, 0x02, 0x0c, 0x6c, 0x63, 0x6d, 0x73, 0x02, 0x10, 0x00, 0x00,
+                                     0x6d, 0x6e, 0x74, 0x72, 0x52, 0x47, 0x42, 0x20, 0x58, 0x59, 0x5a, 0x20 };
+static const size_t sampleICCSize = sizeof(sampleICC) / sizeof(sampleICC[0]);
+
+// Exif bytes are partially checked by libavif. This is a truncated widespread Exif metadata chunk.
+static const uint8_t sampleExif[] = { 0xff, 0x1,  0x45, 0x78, 0x69, 0x76, 0x32, 0xff, 0xe1, 0x12, 0x5a, 0x45,
+                                      0x78, 0x69, 0x66, 0x0,  0x0,  0x49, 0x49, 0x2a, 0x0,  0x8,  0x0,  0x0 };
+static const size_t sampleExifSize = sizeof(sampleExif) / sizeof(sampleExif[0]);
+
+// XMP bytes are not checked by libavif so the content does not matter.
+// This is a truncated widespread XMP metadata chunk.
+static const uint8_t sampleXMP[] = { 0x3c, 0x3f, 0x78, 0x70, 0x61, 0x63, 0x6b, 0x65, 0x74, 0x20, 0x62, 0x65,
+                                     0x67, 0x69, 0x6e, 0x3d, 0x22, 0xef, 0xbb, 0xbf, 0x22, 0x20, 0x69, 0x64 };
+static const size_t sampleXMPSize = sizeof(sampleXMP) / sizeof(sampleXMP[0]);
+
+//------------------------------------------------------------------------------
+// TODO: Move these functions to a tests/common/helper.c shared with avifgridapitest.c
+
+// Fills a plane with a repeating gradient.
+static void fillPlane(int width, int height, int depth, uint8_t * row, uint32_t rowBytes)
+{
+    assert(depth == 8 || depth == 10 || depth == 12); // Values allowed by AV1.
```

Comment:
Note: In the past I've asked people to parenthesize subexpressions when comparisons are involved, to imitate the style used by Joe Drago. For example, in that style this line would be written as follows:
```
    assert((depth == 8) || (depth == 10) || (depth == 12)); // Values allowed by AV1.
```


Since it is easy to forget to do this, Joe and I have decided that we should be more lenient about this.